### PR TITLE
feat: bump to require node > 18

### PR DESCRIPTION
**Pull request by madpah**
BREAKING-CHANGE:

- Updated to require `node>=18`
- Updated a bunch of dependencies to be safer
- Removed dead dependencies
- Resolved some typing errors under later TS standards
- Added LICENSE file

fix: added missing LICENSE file

fix: cleaned unused modules (ran `yarn autoclean`)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -100,7 +100,7 @@ jobs:
           template: basic_fail_1
 
 workflows:
-  version: 2.1
+  version: 2
```
Comment (bhamail) [inline .circleci/config.yml:103]: Any reason for dropping back to version 2 from 2.1? I think 2.1 is needed to run local builds (which still work fine with docker under linux).

Comment (madpah) [reply .circleci/config.yml:103]: I was informed that 2.1 was an invalid value under the schema @bhamail - only reason I changed it!

Comment (madpah) [reply .circleci/config.yml:103]: See https://circleci.com/docs/configuration-reference/#workflows

**Comment by madpah** [comment]
@bhamail @ctownshend @maurycupitt - this PR should now be ready for review

File: .circleci/config.yml (hunk 1 above)
Comment (bhamail) [reply .circleci/config.yml:103]: Odd, and confusing: https://circleci.com/docs/configuration-reference/#version

Later, I'll try to build locally under linux with the 2 value.

**Review by maurycupitt — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (madpah) [reply .circleci/config.yml:103]: Note that is the workflow version, not the file version - which is `2.1`

**Pull request merged**